lstchain/scripts/lstchain_r0_to_r0g.py

```diff
@@ -0,0 +1,235 @@
+#!/usr/bin/env python
+"""
+Script to apply gain selection to the raw fits.fz produced by EvB6 in CTAR1
+format (aka R1v1).
+
+Gain selection is only applied to shower events (not to interleaved pedestal
+and flatfield)
+
+It uses heuristic identification of the interleaved flatfield (FF) events (
+given the occasional problems we have with the FF event tagging)
+
+"""
+import logging
+import protozfits
+import argparse
+import sys
+import re
+
+from pathlib import Path
+from ctapipe.containers import EventType
+
+import lstchain.paths as paths
+import numpy as np
+from contextlib import ExitStack
+
+parser = argparse.ArgumentParser(description="Gain selector program (R0 to "
+                                             "R0G)")
+parser.add_argument('-f', '--R0-file', dest='input_file',
+                    type=str, help='Input R0 file name (of stream 1)')
+
+parser.add_argument('-o', '--output-dir', dest='output_dir',
+                    type=str, default='./',
+                    help='Output directory')
+
+parser.add_argument('--log', dest='log_file',
+                    type=str, default=None,
+                    help='Log file name')
+
+parser.add_argument('--no-flatfield-heuristic', action='store_const', 
+                    const=False, 
+                    dest="use_flatfield_heuristic", default=True,
+                    help=("If given, do *not* identify flatfield events"
+                          " heuristically from the raw data. "
+                          "Trust event_type."))
+
+# Range of waveform to be checked (for gain selection & heuristic FF
+# identification)
+SAMPLE_START = 3
+SAMPLE_END = 39
+
+# Level of high gain (HG) required for switching to low gain (LG)
+THRESHOLD = 3900
+
+# Events for which gain selection will be applied:
+EVENT_TYPES_TO_REDUCE = [EventType.SUBARRAY, EventType.UNKNOWN]
+
+def main():
+    args = parser.parse_args()
+  
+    input_file = args.input_file
+    output_dir = args.output_dir
+    use_flatfield_heuristic =  args.use_flatfield_heuristic
+
+    log = logging.getLogger(__name__)
+    log.setLevel(logging.INFO)
+
+    log_file = args.log_file
+    runinfo = paths.parse_r0_filename(input_file)
+    if log_file is None:
+        log_file = f'R0_to_R0g_Run{runinfo.run:05d}.{runinfo.subrun:04d}.log'
+
+    formatter = logging.Formatter('%(asctime)s - '
+                                  '%(levelname)s - %(message)s',
+                                  '%Y-%m-%d %H:%M:%S')
+    handler = logging.FileHandler(log_file, mode='w')
+    handler.setFormatter(formatter)
+    logging.getLogger().addHandler(handler)
+
+    # Loop over the files (4 streams) to perform the gain selection:
+    
+    input_stream_names = [Path(Path(input_file).parent,
```

Comment:
Since each stream is now independent, do we want to have an internal loop here? Why not just call the script for each input file?

Reply:
I see no obvious advantage, and I think it is cleaner to keep the whole subrun as the input here.
Imagine we eventually want to e.g. check the rate of interleaveds, or something else which needs access to go over all the events.